### DISCUSSION TITLE
Prisma: aggregating values from child table in parent

### DISCUSSION THREAD

**Discussion by msreekm** [prisma/prisma#14119] [Q&A]
Hi 
what is the recommended apprach to aggregating values in parent table ?

1. Prisma middleware
2. Database Triggers separate from prisma schema
3. use prisma transactions that aggregate from child data  and update parent table
4. any other approach?

**Comment by austincrim** [comment]
Hey @msreekm 👋,

Just to make sure we are on the same page, could you give me an example of what you are trying to accomplish?

**Reply by msreekm** [reply]
example-  
Order Table -store order info
Order Line Item - store each line item has price column

in Order table ,I would like to have an total price column based on sum of price columns in order line table.  @austincrim 
 
Do prisma users also write database triggers and include in migration script? is this a common pattern?

**Reply by austincrim** [reply]
There is no way to automatically compute the column based on the child table, but you could manually do it in an [interactive transaction](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#interactive-transactions-in-preview) as you mentioned.

It is a common pattern to customize the Prisma-generate migrations with custom SQL, including triggers, indexes, and other unsupported features.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
